sigma/plugins/fun/xkcd.py

```diff
@@ -2,24 +2,36 @@
 import aiohttp
 import discord
 
-
 async def xkcd(cmd, message, args):
 
-    comic_no = str(random.randint(1, 2171))
+    random_comic = True
     if args:
         if not is_int(args[0]):
             await message.channel.send(None, embed=discord.Embed(title=':exclamation: Invalid number', color=0x993333))
             return
-        
+
         comic_no = args[0]
-        
-    joke_url = 'http://xkcd.com/' + comic_no + '/info.0.json'
-    async with aiohttp.ClientSession() as session:
-        async with session.get(joke_url) as data:
-            try: joke_json = await data.json()
-            except Exception:
-                await message.channel.send(None, embed=discord.Embed(title=':exclamation: Requested xkcd doesn\'t exist', color=0x993333))
-                return
+        random_comic = False
+
+    while True:
```

Comment:
What's the while loop for?

Reply:
If the user does not input any arguments, it is necessary to request the URL twice: once to get the latest comic number, and once to actually get the comic that is going to be displayed. The loop condenses that process. At most, the code in the loop runs twice.

Reply:
That's also why I have the `random_comic` flag; if the flag is `True`, the loop runs normally, and I can then change it to `False` in Line 30, so when the loop runs again, it can quit out.

Reply:
The while loop makes it more convoluted

Reply:
Without it, I would have two almost identical situations for when the user requests a specific comic and when a random comic is retrieved. The `while` loop saves code.

Reply:
Get rid of it and we'll work from there

Reply:
124ab64

Looking at this now, everything in

```python
async with aiohttp.ClientSession() as session:
    async with session.get(joke_url) as data:
        try: joke_json = await data.json()
        except Exception:
            await message.channel.send(None, embed=discord.Embed(title=':exclamation: Requested xkcd doesn\'t exist', color=0x993333))
            return
```
could probably go in a function. What do you think?

Reply:
Function will add even more nesting

Reply:
So just leave it as is?